### PR TITLE
RELATED: RAIL-3593 PivotTable ended-up in infinite loading loop, plug viz callback are not updated

### PR DESCRIPTION
**Pull request by ivanmjartan**
---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4715/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6342/ : FAILURE in 4m 32s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4717/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6344/ : SUCCESS in 18m 41s

File: libs/sdk-ui-ext/src/insightView/InsightRenderer.tsx [hunk 1]
```diff
@@ -247,17 +247,65 @@ class InsightRendererCore extends React.PureComponent<IInsightRendererProps & Wr
 
 export const IntlInsightRenderer = compose(injectIntl, withTheme, withContexts)(InsightRendererCore);
 
+/**
+ * Since InsightRendererImpl only sets callback (onDrill or pushData etc) on the first render (this is a PlugVis API, there is no way to update onDrill there),
+ * we have to update callback manually by this hook.
+ *
+ * This problem leads to unpredictable closures problems in using react hooks effect and dependencies arrays.
```
Comment (kandl) [inline libs/sdk-ui-ext/src/insightView/InsightRenderer.tsx:254]: If I understand it correctly, the issue was more about the fact, that the changed callbacks (with different reference) were not properly propagated to the "visualization" instance (as it took only the first provided callbacks).

Comment (ivanmjartan) [reply libs/sdk-ui-ext/src/insightView/InsightRenderer.tsx:254]: Yes you are right

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1596

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1331/ : SUCCESS in 1m 04s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1607/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/385/ : SUCCESS in 11m 42s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6348/ : SUCCESS in 14m 15s

**Pull request merged**